plugins/xml-extractor/src/index.ts

```diff
@@ -0,0 +1,71 @@
+import type { Client } from '@flatfile/listener'
+import { fileBuffer } from '@flatfile/plugin-file-buffer'
+import { flatToValues, schemaFromObjectList, xmlToJson } from './parser'
+import api, { Flatfile } from '@flatfile/api'
+
+export const XMLExtractor = (opts?: {
+  separator?: string
+  attributePrefix?: string
+  transform?: (row: Record<string, any>) => Record<string, any>
+}) => {
+  return (client: Client) => {
+    client.use(
+      fileBuffer('.xml', async (file, buffer, event) => {
+        const job = await api.jobs.create({
+          type: 'file',
+          operation: 'extract',
+          status: 'ready',
+          source: event.context.fileId,
+        })
+        try {
+          const json = xmlToJson(buffer.toString()).map(
```

Comment:
Invocation of `xmlToJson(..).map` looks like something unsual is going on here. Better to clarify naming here, like `xmlToJsonArray`.